### PR TITLE
Extend validator to allow simple error templates

### PR DESCRIPTION
**Pull request by nicolaasuni**

### CODE REVIEW

File: pkg/validator/validation_error.go [hunk 1]
```diff
@@ -0,0 +1,56 @@
+package validator
+
+// ValidationError is a custom error adding a Field member.
+type ValidationError struct {
```
Comment (mgiaccone) [inline pkg/validator/validation_error.go:4]: The `Validation` prefix seems redundant. When used it becomes `validator.ValidationError` when `validator.Error` would be enough

Comment (mgiaccone) [reply pkg/validator/validation_error.go:4]: Also the filename could be just `error.go`

File: pkg/validator/validator.go [hunk 2]
```diff
@@ -78,29 +38,46 @@ func New(opts ...Option) (*Validator, error) {
 }
 
 // ValidateStruct validates the structure fields tagged with "validate".
-func (v *Validator) ValidateStruct(obj interface{}) error {
-	err := v.V.Struct(obj)
-	if err == nil {
+func (v *Validator) ValidateStruct(obj interface{}) (err error) {
+	verr := v.V.Struct(obj)
```
Comment (mgiaccone) [inline pkg/validator/validator.go:42]: maybe `vErr`?

File: pkg/validator/validator_test.go [hunk 3]
```diff
@@ -30,18 +30,27 @@ func TestNew(t *testing.T) {
...
 				WithFieldNameTag("test_tag"),
+				WithDefaultTranslations(),
 				WithValidationTranslated("test_tag", nil, nil, nil),
 			},
 			want:    false,
 			wantErr: true,
 		},
 		{
-			name: "success with opts applied",
+			name: "success with default tranaslations",
```
Comment (mgiaccone) [inline pkg/validator/validator_test.go:41]: typo, it should be `translations`

File: pkg/validator/validation_error.go (hunk 1 above)
Comment (nicolaasuni) [reply pkg/validator/validation_error.go]: fixed

File: pkg/validator/validator.go (hunk 2 above)
Comment (nicolaasuni) [reply pkg/validator/validator.go]: fixed

File: pkg/validator/validator_test.go (hunk 3 above)
Comment (nicolaasuni) [reply pkg/validator/validator_test.go]: fixed

**Review by mgiaccone — DISMISSED** [review]

**Review by mgiaccone — APPROVED** [review]

**Review by conves — APPROVED** [review]

**Pull request merged**